crypto/keymanager.go

```diff
@@ -0,0 +1,113 @@
+package crypto
+
+import (
+	"fmt"
+
+	"github.com/tbd54566975/web5-go/crypto/dsa"
+	"github.com/tbd54566975/web5-go/jwk"
+)
+
+// KeyManager is an abstraction that can be leveraged to manage/use keys (create, sign etc) as desired per the given use case
+// examples of concrete implementations include: AWS KMS, Azure Key Vault, Google Cloud KMS, Hashicorp Vault etc
+type KeyManager interface {
```

Comment:
```
                          KeyManager (interface)
                              ^
                        LocalKeyManager (implements interface?) 
                  ^           ^                       ^
   dsa.GetPublicKey() dsa.GeneratePrivateKey()  dsa.Sign() 
(called by LocalkeyManager methods of same name
 turns around and calls either eddsa or ed25519 methods)
```

thit is my mental model - is this correct or am i off?

Reply:
is there like a `ClassABC implements InterfaceXYZ` kind of lingo to enforce required method implementation in go?

Reply:
correct! and something like `AWSKeyManager` wouldn't call `dsa.*` and instead make requests directly to AWS KMS to create and store said key. if the algorithm isn't supported by the concrete implementation, an error would be returned